### DISCUSSION TITLE
Pinia store doesn't work

### DISCUSSION THREAD

**Discussion by romanslonov** [histoire-dev/histoire#151] [Q&A]
Hello, I did a mimial reporpoduction what is actually doesn't work for me. I define a pinia store for modal component and then trying to open the modal in story but store doesn't seems to work properly, values are not changing. Maybe I am doing something wrong? Help please.

I also have a warn in console about so maybe there is an issue with initialization: 
```
[Vue warn]: injection "Symbol(pinia)" not found. 
  at <Modal.story story= {id: 'src-stories-modal-story-vue', title: 'Modal', group: null, layout: {…}, icon: null, …} > 
```

Repo: https://github.com/romanslonov/histoire-pinia-issue

**Comment by victorlmneves** [comment]
@romanslonov I have used Pinia on one component and it's working, but I didn't use the global config, I have set up it on the story

```
function mySetupApp ({ app }) {
  app.use(createPinia());
}
```

```
<Story group="form" title="Components/Form/FormValidationScope" :setup-app="mySetupApp">
```

Try this way to see if works

**Comment by hrishikesh-k** [comment]
@victorlmneves Unfortunately, that doesn't work and it thrown an error:

```
getActivePinia was called with no active Pinia. Did you forget to install pinia?
```

Do you happen to have an example handy where you had got it working successfully?

**Reply by victorlmneves** [reply]
can you create a stablitz example?
I don't have any public example

**Reply by hrishikesh-k** [reply]
Here's a reproduction: https://stackblitz.com/edit/hk-histoire-d-151. `npm run story:dev` should show the error.

**Reply by victorlmneves** [reply]
Here is an example based on your code working fine
I saw some weird code. Why are using ref on the store? The store is reactive
https://stackblitz.com/edit/hk-histoire-d-151-br8kf8

**Reply by hrishikesh-k** [reply]
The weird code is valid based on Pinia's documentation on "Setup Stores": https://pinia.vuejs.org/core-concepts/#setup-stores

Your example has removed the core functionality I need in my app - that is to read and manipulate the store from within the story. As soon as I call `useButtonStory()` in the story, it fails with the same error.

I'm currently using the workaround I mentioned here: https://github.com/histoire-dev/histoire/discussions/260#discussioncomment-5711691, because otherwise, I don't see this could be working.

**Reply by victorlmneves** [reply]
Where do you get the error? For me, it works totally fine running the app and running Histoire

**Reply by victorlmneves** [reply]
Here's a working version with the store as you have
https://stackblitz.com/edit/hk-histoire-d-151-knv7xa

**Reply by hrishikesh-k** [reply]
I'm not sure how you're not seeing the error. I literally just uncommented the line in `v-button.story.vue` in your latest link and I see it happen:

<img width="797" alt="image" src="https://user-images.githubusercontent.com/11403551/234848770-d1377c56-96fe-4d34-ab7e-7bcaa92e6d61.png">


To be able to use the store in the Story, calling `useButtonStore()` is required, or else it's as good as not using Pinia.

**Reply by victorlmneves** [reply]
The line you uncomment it's not required as it's part of the component, and to be honest I didn't get why you want that paragraph

**Reply by victorlmneves** [reply]
on both examples that I have provided you have access to the store inside the story

**Reply by hrishikesh-k** [reply]
I provided the exact reproduction of why I need that line. In the story, I need to see the count as well as within the actual component. If you've any other way of accessing the data inside the story, feel free to let me know. If not, based on the other discussion I shared, this is indeed an issue with Histoire.

The only reason I participated in this discussion was because I'm not able to share the state of the store in the story as well as the actual component. I can create an instance of the store in both, but both those instances are not linked to each other, that is, changes made in one are not visible in the other. This was fairly evident in the reproduction I shared in the beginning. To explain it again (refer to my original reproduction):

1. In `v-button.story.vue` on line 15, I have: `<p>Count in story: {{vButtonStore.count}}</p>`
2. In `v-button.vue`, on line 10, I have: `<p>Count in component: {{vButtonStore.count}}</p>`

I need both of those to stay in-sync with the value in the store. My problem is not only being able to pass Pinia, but to also read data and make changes to it from the component as well as the story. Without this, I don't see why someone would need to pass the store.

Maybe your code works, but I don't know how I can access the data from the store without calling `useButtonStore()`.

**Reply by victorlmneves** [reply]
That doesn't work even if you set pinia globally, because then you get a kind of stores for the count, one from within story and the generic one
Here is an example of what you did without the pinia error, but as can see, you increment it using the button but the value form the one you set on the story doesn't get updated as it's a "different store" that you set directly on the story
https://stackblitz.com/edit/hk-histoire-d-151-j9ce7p

**Reply by victorlmneves** [reply]
In all examples I provide you have access to store directly on the story by importing the component that has that functionality and to be honest I'm not getting why you are adding that paragraph and the need of "2 stores"

**Reply by hrishikesh-k** [reply]
I'm sorry, I'm sure it's me, but there's probably some kind of mis-communication happening due to which I'm unable to explain myself properly. As of now, I'll continue with my workaround linked earlier. Thanks for chiming in.
